app/controllers/api/v1/stats/referral_codes_controller.rb

```diff
@@ -0,0 +1,37 @@
+class Api::V1::Stats::ReferralCodesController < Api::V1::StatsController
+    def index
+      data = PromoRegistration.all.map { |promo_registration| promo_registration.referral_code }
```

Comment:
Preferably use `pluck` as it's more readable and probably more performant.

e.g. `data = PromoRegistration.pluck(:referral_code)`

Reply:
Read up on the doc, pluck is indeed a better method to use here. Thanks @yachtcaptain23